### PR TITLE
feat(Wasm,Skia,iOS): Support FontStretch for TextBlock and TextElement

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): Part of #7614

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

`FontStretch` not supported.

## What is the new behavior?

Skia GTK:

image

Skia WPF:

image

Wasm:

image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/FuncMemoizeExtensions.cs [hunk 1]
```diff
@@ -191,6 +191,35 @@ public static Func<T> AsMemoized<T>(this Func<T> func)
...
+		/// </summary>
+		/// <typeparam name="TParam1">The first parameter type to memoize</typeparam>
+		/// <typeparam name="TParam2">The second parameter type to memoize</typeparam>
+		/// <typeparam name="TParam3">The third parameter type to memoize</typeparam>
+		/// <typeparam name="TParam4">The fourth parameter type to memoize</typeparam>
+		/// <typeparam name="TParam5">The fifth parameter type to memoize</typeparam>
+		/// <typeparam name="TParam6">The sixth parameter type to memoize</typeparam>
+		/// <param name="func">the function to evaluate</param>
+		/// <returns>A memoized value</returns>
+		public static Func<TParam1, TParam2, TParam3, TParam4, TParam5, TParam6, TResult> AsMemoized<TParam1, TParam2, TParam3, TParam4, TParam5, TParam6, TResult>(this Func<TParam1, TParam2, TParam3, TParam4, TParam5, TParam6, TResult> func)
```
Comment (jeromelaban) [inline src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/FuncMemoizeExtensions.cs:205]: Why do you need this in this PR ?

Comment (Youssef1313) [reply src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/FuncMemoizeExtensions.cs:205]: @jeromelaban In UIFontHelper.iOS.cs, `_tryGetFont` is now 6 parameters, and we call `AsMemoized` on it.

Comment (jeromelaban) [reply src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/FuncMemoizeExtensions.cs:205]: Ah! Missed it. Thanks. Maybe we should be using a record as the parameter then?

Comment (Youssef1313) [reply src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/FuncMemoizeExtensions.cs:205]: We could. We could also use a tuple like in:

https://github.com/unoplatform/uno/blob/d83450c5a1559f2532b7b2c00f77edfd9b2b0b92/src/Uno.UI/Controls/UIStringAttributesHelper.iOS.cs#L14-L27

Let me know your preference.

Comment (jeromelaban) [reply src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/FuncMemoizeExtensions.cs:205]: A tuple can work as well, though having a record makes for less duplication (you'll have to specify the Tuple at least twice).

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs [hunk 2]
```diff
@@ -65,5 +68,56 @@ public async Task When_Null_FontFamily()
...
+				var point = textBlock.TransformToVisual(screenshot.RenderedElement).TransformPoint(new Point(x, textBlock.ActualHeight / 2));
+				var pixel = screenshot.GetPixel((int)point.X, (int)point.Y);
+				if (ImageAssert.AreSameColor(Windows.UI.Colors.Blue, pixel, tolerance: 10, out _))
+				{
+					return x;
+				}
+			}
+
+			return -1;
+		}
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs:121]: I don't think this is commonly needed, so I didn't add it to `ImageAssert`.

**Comment by Youssef1313** [comment]
Test is failing on Skia WPF in CI, but is passing locally 😕 

For iOS, test is failing, and I can't see what's going on as I don't have a device.

For Wasm, test is disabled because `RenderTargetBitmap` doesn't seem to be implemented, but from the manual test, things are working well

https://github.com/unoplatform/uno/blob/fe91406526782b70f86273149f1c291903afa5f9/src/Uno.UI/UI/Xaml/Media/Imaging/RenderTargetBitmap.cs#L2-L4

**Comment by Youssef1313** [comment]
@jeromelaban I'm not sure how can I move forward with this PR :(

**Comment by jeromelaban** [comment]
There's at least the Skia tests that you can validate: https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1983794&_a=resultSummary&resultId=104292

**Comment by jeromelaban** [comment]
As for iOS failing, you can take a look at the screenshots here:
https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1983788&_a=resultSummary&resultId=100000 (the Tear_down_on_error.png one). The interesting part is that the text should show on the right side of the screen but does not. (and there's a weird text superposition on the left side of the screen).

**Comment by MartinZikmund** [comment]
@Youssef1313 Is this still relevant or should be closed as it was superseeded with other changes later?

**Pull request closed without merging**